flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkUtils.java

```diff
@@ -243,22 +244,21 @@ public static void deleteJobGraphInKubernetesHA(
                         .withLabels(haConfigMapLabels)
                         .list();
 
-        configMaps
-                .getItems()
-                .forEach(
-                        configMap -> {
-                            final boolean isDeleted =
-                                    configMap
-                                            .getData()
-                                            .entrySet()
-                                            .removeIf(FlinkUtils::isJobGraphKey);
-                            if (isDeleted) {
-                                LOG.info(
-                                        "Job graph in ConfigMap {} is deleted",
-                                        configMap.getMetadata().getName());
-                            }
-                        });
-        kubernetesClient.resourceList(configMaps).inNamespace(namespace).createOrReplace();
+        boolean shouldUpdate = false;
+        for (ConfigMap configMap : configMaps.getItems()) {
+            if (configMap.getData() == null || configMap.getData().isEmpty()) {
+                continue;
+            }
+            final boolean isDeleted =
```

Comment:
I am afraid we could not set `shouldUpdate` here since it might be overridden to false by subsequent loop wrongly.

BTW, the reason why I do not break the loop when `isDeleted` is true is we could have more than one ConfigMaps need to remove jobgraph when upgrading Flink version from 1.14 to 1.15.

WDYT?

Reply:
Oh, yes. I misunderstand it, I ignore that it's in a loop :( .     +1 for this

Reply:
Thanks for the confirmation.